config/checkstyle-examples-checks.xml

```diff
@@ -61,12 +50,50 @@
     <property name="minimum" value="1"/>
     <property name="maximum" value="1"/>
   </module>
+  <module name="RegexpSingleline">
+    <property name="id" value="forbiddenRegexpInViolationMessage"/>
+    <property name="format" value="^.*//.* violation.*(\.\*|\.\+).*$"/>
+    <property name="message"
+              value="Forbidden usage of regex: .* and .+ are not allowed"/>
+    <property name="maximum" value="0"/>
+  </module>
+  <module name="RegexpSingleline">
+    <property name="id" value="discouragedRegexpInViolationMessage"/>
+    <property name="format" value="^.*//.* violation.*\\[\.\+\*\?\|\{\}\(\)\[\]\^\-\$].*$"/>
+    <property name="message"
+              value="Discouraged usage of escaped special regex symbols. Please ensure violation
+                     message really needs this symbol"/>
+    <property name="maximum" value="0"/>
+  </module>
+  <module name="RegexpSingleline">
+    <property name="id" value="presenceOfCompilationComment"/>
+    <property name="format" value="^//non-compiled with javac: Compilable with Java(\d+)"/>
+    <property name="message"
+              value="Missing compilation comment"/>
+    <property name="minimum" value="1"/>
+    <property name="maximum" value="1"/>
+  </module>
+
+  <!-- Miscellaneous -->
+  <module name="NewlineAtEndOfFile"/>
 
+  <!-- Size -->
+  <module name="FileLength">
+    <property name="max" value="45"/>
+  </module>
   <module name="LineLength">
     <property name="max" value="85"/>
   </module>
 
+  <!-- Whitespace -->
+  <module name="FileTabCharacter">
+    <property name="eachLine" value="false"/>
+  </module>
+
   <module name="TreeWalker">
+
+    <!-- Miscellaneous -->
+    <module name="CommentsIndentation"/>
```

Comment:
If this one is weird, let's not be afraid to deactivate it if it causes us pain.

Reply:
No issues for now. Will keep that in mind.